jep/1/README.adoc

```diff
@@ -0,0 +1,939 @@
+  will not be approved as a JEP Draft.
+. **Reasoning** - The reasoning describes why particular design decisions were made.
+  It should describe alternate designs that were considered and related work,
+  e.g. how the feature is supported in other systems.
++
+The reasoning should provide evidence of consensus within the
+community and discuss important objections or concerns raised
+during discussion.
+
+. **Backwards Compatibility** -
+  All JEPs must include a section describing any incompatibilities and their severity.
+  The JEP must explain how the sponsor proposes to deal with these incompatibilities.
+  If there are no dackwards compatibility concerns, the section must say that.
+  JEP submissions that do not adequately discuss backwards compatibility,
+  when such discussion is required will not be approved as JEP Drafts.
+. **Reference Implementation** --
+  If a JEP will include code changes, an open source reference implementation
+  must be completed before any JEP is given status "Final".
+  The reference implementation need not be completed before the JEP is accepted.
+  While there is merit to the approach of reaching consensus on the
+  specification and reasoning before writing code,
+  the principle of "rough consensus and running code" is still useful
+  when it comes to resolving many discussions of API details.
+. **References** -- When moving a JEP from a Draft to "Accepted" or "Final" state,
+  the references section should be updated to include links to the pull requests
+  and mailing list discussions which were involved in the process. The JEP
+  should self-document the process in which it was developed.
+
+The final implementation must include test code and documentation
+appropriate for either the Jenkins user or developer documentation.
+
+==== JEP File Format and Style
+
+JEPs are UTF-8 encoded text files using the
+link:https://asciidoctor.org[AsciiDoc] format.
+AsciiDoc allows for rich markup that is still quite easy to read,
+but also results in good-looking and functional HTML.
+
+The following are style guidelines for JEP asciidoc files.
+Items earlier in the list are higher importance than later items.
+
+===== Keep `link` on one line (REQUIRED)
+
+Asciidoc is newline insensitive by default.
+This means that structures such as links can wrap across lines, like this:
+
+[source, asciidoc]
+--
+link:http://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line[One
+sentence per line]
+--
+
+This does not improve readability or diffing.
+Keep each link all on the same line.
+
+[source, asciidoc]
+--
+link:http://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line[One sentence per line]
+--
+
+===== 120 characters per line (REQUIRED)
+
+Aside from the structures mentioned above this point,
+lines MUST be wrapped before 120 characters per line.
+This is required to make diffs and comments reasonably accurate and specific.
```

Comment:
This doesn't make any sense. The rendered diff on GH is a useless mess, and the raw diff is even worse.

One line per sentence is clearly the only sensible format. Screen too small? Enable line wrap.

It even assists writing: Line gets too long? Likely a run-on sentence that needs fixing.

Reply:
@daniel-beck  

> This doesn't make any sense. The rendered diff on GH is a useless mess, and the raw diff is even worse.

What do you mean here?  The diff for this PR?  Yes, it is insane.  We've made sweeping changes.  

> One line per sentence is clearly the only sensible format. Screen too small? Enable line wrap.

> It even assists writing: Line gets too long? Likely a run-on sentence that needs fixing.

We were avoiding putting any formatting rules in to avoid exactly this situation - nitpicking about which rules are needed or not.  You have a system that you think is best, I have a slightly different one, tyler has another, etc.  

In my experience people with line wrap do not add newlines.  Which then means I as reviewer have to tell them to do so. Also, there are plenty of cases where per sentence breaks result in very long lines and breaking on commas doesn't improve matters (this sentence, for example).  

I chose these rules specifically because they force people to add newline more often.  In my experience, that tends to result in easier diffs.  In the case where the diffs are messy, it indicates lots of change.  

If you'd like to propose some alternative, maybe run some experiments, I'm completely open to changing these.